src/opendr/perception/panoptic_segmentation/efficient_ps/efficient_ps_learner.py

```diff
@@ -0,0 +1,521 @@
+            meta=meta
+        )
+
+        timestamp = time.strftime('%Y%m%d_%H%M%S', time.localtime())
+        runner.timestamp = timestamp
+        runner.register_training_hooks(self._cfg.lr_config, self._cfg.optimizer_config,
+                                       self._cfg.checkpoint_config, self._cfg.log_config)
+
+        if val_dataset is not None:
+            val_dataset.pipeline = self._cfg.test_pipeline
+            val_dataloader = build_dataloader(
+                val_dataset.get_mmdet_dataset(test_mode=True),
+                imgs_per_gpu=1,
+                workers_per_gpu=self.num_workers,
+                dist=False,
+                shuffle=False
+            )
+            runner.register_hook(EvalHook(val_dataloader, interval=1, metric=['panoptic']))
+
+        runner.run(dataloaders, self._cfg.workflow, self.iters)
+        self._is_model_trained = True
+
+    def eval(self,
+             dataset: Any,
+             print_results: bool=False
+             ) -> Dict[str, Any]:
+        """
+        This method is used to evaluate the algorithm on a dataset and returns the following stats:
+            - Panoptic Quality (PQ)
+            - Segmentation Quality (SQ)
+            - Recognition Quality (RQ)
+
+        :param dataset: Object that holds the evaluation dataset
+        :type dataset: Dataset class type
+        :param print_results: If set to True, the computed metrics will be printed to STDOUT
+        :type print_results: bool
+        :return: Returns stats regarding the evaluation
+        :rtype: dict
+        """
+        sampler = SequentialSampler(dataset)
+        dataloader = DataLoader(dataset, batch_size=self.batch_size, sampler=sampler, num_workers=self.num_workers,
+                                collate_fn=lambda x: x)
+
+        with tqdm(dataloader, unit='batch', desc='Evaluating...') as pbar:
+            for i, batch in enumerate(dataloader):
+                images = [data[0] for data in batch]
+                predictions = self.infer(images, return_raw_logits=True)
+                save_panoptic_eval_(predictions, path=self.temp_path)
+                pbar.update(1)
+
+        results = dataset.evaluate(os.path.join(self.temp_path, 'tmp'), os.path.join(self.temp_path, 'tmp_json'))
+
+        if print_results:
+            msg = f"{'Category':<14s}| {'PQ':>5s} {'SQ':>5s} {'RQ':>5s} {'N':>5s}\n"
+            msg += "-" * 41 + "\n"
+            for x in ['All', 'Things', 'Stuff']:
+                msg += f"{x:<14s}| {results[x]['pq'] * 100:>5.1f} {results[x]['sq'] * 100:>5.1f} "
+                msg += f"{results[x]['rq'] * 100:>5.1f} {results[x]['n']:>5d}\n"
+            msg += "-" * 41 + "\n"
+            for cat, value in results['per_class'].items():
+                msg += f"{cat:<14s}| {value['pq'] * 100:>5.1f} {value['sq'] * 100:>5.1f} {value['rq'] * 100:>5.1f}\n"
+            msg = msg[:-1]
+            print(msg)
+
+        shutil.rmtree(self.temp_path)
+        return results
```

Comment:
Are these results a dictionary (eval should return a dictionary according to specs)?

Reply:
Yes, as specified in the function's type hints, the returned `results` are a dictionary comprising panoptic, segmentation, and recognition qualities.